src/main/java/org/saltyrtc/tasks/webrtc/messages/Candidate.java

```diff
@@ -10,33 +10,79 @@
 
 import org.saltyrtc.client.annotations.NonNull;
 import org.saltyrtc.client.annotations.Nullable;
+import org.saltyrtc.client.exceptions.ValidationError;
+import org.saltyrtc.client.helpers.ValidationHelper;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
 
 /**
  * Immutable POJO containing candidate information.
  */
-class Candidate {
-    @NonNull
-    private final String candidate;
-    @Nullable
-    private final String sdpMid;
-    @Nullable
-    private final Integer sdpMLineIndex;
-
-    public Candidate(@NonNull String candidate, @Nullable String sdpMid, @Nullable Integer sdpMLineIndex) {
+public class Candidate {
+    @NonNull private static final String FIELD_CANDIDATE = "candidate";
+    @NonNull private static final String FIELD_SDP_MID = "sdpMid";
+    @NonNull private static final String FIELD_SDP_M_LINE_INDEX = "sdpMLineIndex";
+
+    @NonNull private final String candidate;
```

Comment:
Should we call this field `sdp` to match the WebRTC `IceCandidate` constructor?

    public IceCandidate(String sdpMid, int sdpMLineIndex, String sdp)

Reply:
If anything, should probably be renamed to `candidateSdp`.